modules/apps/batch-engine/batch-engine-service/src/main/java/com/liferay/batch/engine/internal/util/ItemIndexThreadLocal.java

```diff
@@ -16,32 +16,38 @@
 
 import com.liferay.petra.lang.CentralizedThreadLocal;
 
-import java.util.HashMap;
-import java.util.Map;
+import java.util.LinkedList;
+import java.util.Queue;
 
 /**
  * @author Matija Petanjek
  */
 public class ItemIndexThreadLocal {
 
-	public static int get(Object item) {
-		Map<Object, Integer> itemIndexMap = _itemIndexMap.get();
+	public static void add(int itemIndex) {
+		Queue<Integer> indexQueue = _indexQueue.get();
 
-		return itemIndexMap.get(item);
+		indexQueue.add(itemIndex);
 	}
 
-	public static void put(Object item, int itemIndex) {
-		Map<Object, Integer> itemIndexMap = _itemIndexMap.get();
+	public static void clear() {
+		_indexQueue.remove();
+	}
+
+	public static int get() {
```

Comment:
Brian usually renames this to match the method call on the Collection (peek()), but even if we don't change it I don't think he will reject the PR because of it.